sources/Algo/src/Algo.cpp

```diff
@@ -115,6 +115,7 @@ GeneratorDefinitionAlgorithm::operator()(const NodePtr& node) {
     if ((model == GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN || model == GeneratorDefinition::ModelType::REMOTE_DIAGRAM_PQ_SIGNALN ||
          model == GeneratorDefinition::ModelType::PROP_DIAGRAM_PQ_SIGNALN || model == GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN) &&
         !isDiagramValid(generator)) {
+      busesWithDynamicModel_.erase(generator.regulatedBusId);
```

Comment:
in worst case, we have an insert then an erase, which is more time consuming than performing twice the test (model == ...) in switch case